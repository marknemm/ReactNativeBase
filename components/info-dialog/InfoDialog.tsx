import { Dialog } from '@rneui/themed';
import type { InfoDialogProps } from './InfoDialog.interfaces';

/**
 * A common dialog that displays information.
 *
 * Contains a title and a close button.
 *
 * @param props The component {@link InfoDialogProps}.
 * @returns The {@link InfoDialog} component.
 */
const InfoDialog: React.FC<InfoDialogProps> = ({
  children,
  isVisible = false,
  onClose,
  title,
  ...dialogProps
}) => (
  <Dialog
    isVisible={isVisible}
    onBackdropPress={onClose}
    {...dialogProps}
  >
    {title && <Dialog.Title title={title} />}
    {children}
    <Dialog.Actions>
      <Dialog.Button onPress={onClose}>Close</Dialog.Button>
    </Dialog.Actions>
  </Dialog>
);

export type * from './InfoDialog.interfaces';
export default InfoDialog;
