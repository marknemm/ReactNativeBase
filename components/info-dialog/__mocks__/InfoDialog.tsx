import type { InfoDialogProps } from '@components/info-dialog/InfoDialog.interfaces';

const MockInfoDialog = jest.fn().mockImplementation((props: InfoDialogProps) => {
  const ActualComponent = jest.requireActual('@components/info-dialog/InfoDialog').default;
  return <ActualComponent {...props} />;
});

export default MockInfoDialog;
