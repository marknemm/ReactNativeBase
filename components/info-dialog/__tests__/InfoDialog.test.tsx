import InfoDialog from '@components/info-dialog/InfoDialog';
import { Text } from '@rneui/themed';
import AppProvider from '@test/contexts/app/AppProvider';
import { render, screen, userEvent } from '@testing-library/react-native';

describe('<InfoDialog />', () => {
  const title = 'Test Title';
  const contentStr = 'Test Content';

  describe('display inputs', () => {
    it('Does not render when not visible', () => {
      render(
        <InfoDialog title={title}>
          <Text>{contentStr}</Text>
        </InfoDialog>,
        { wrapper: AppProvider }
      );

      const titleText = screen.queryByText(title);
      expect(titleText).toBeFalsy();
    });

    it('Renders title', () => {
      render(
        <InfoDialog
          isVisible
          title={title}
        />,
        { wrapper: AppProvider }
      );

      const titleText = screen.queryByText(title);
      expect(titleText).toBeVisible();
    });

    it('Renders children', () => {
      render(
        <InfoDialog isVisible>
          <Text>{contentStr}</Text>
        </InfoDialog>,
        { wrapper: AppProvider }
      );

      const contentText = screen.queryByText(contentStr);
      expect(contentText).toBeVisible();
    });
  });

  describe('on close', () => {
    it('invokes onClose callback', async () => {
      const onClose = jest.fn();
      render(
        <InfoDialog
          isVisible
          onClose={onClose}
        />,
        { wrapper: AppProvider }
      );

      const closeButton = screen.getByRole('button', { name: 'Close' });
      await userEvent.press(closeButton);

      expect(onClose).toHaveBeenCalledTimes(1);
    });
  });

  describe('snapshots', () => {
    it('renders correctly when not visible', () => {
      render(
        <InfoDialog />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly when visible', () => {
      render(
        <InfoDialog isVisible />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with title and children', () => {
      render(
        <InfoDialog
          isVisible
          title={title}
        >
          <Text>{contentStr}</Text>
        </InfoDialog>,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
