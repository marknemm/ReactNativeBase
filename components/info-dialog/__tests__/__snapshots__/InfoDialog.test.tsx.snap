// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoDialog /> snapshots renders correctly when not visible 1`] = `null`;

exports[`<InfoDialog /> snapshots renders correctly when visible 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  replaceTheme={[Function]}
  testID="Internal__Overlay"
  transparent={true}
  updateTheme={[Function]}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, .4)",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    testID="RNE__Overlay__backdrop"
  />
  <View
    pointerEvents="box-none"
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="RNE__Overlay__Container"
  >
    <View
      style={
        {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderRadius": 3,
          "padding": 20,
          "shadowColor": "rgba(0, 0, 0, .3)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowRadius": 4,
          "width": "75%",
        }
      }
      testID="RNE__Overlay"
    >
      <View
        style={
          {
            "flexDirection": "row-reverse",
            "flexWrap": "wrap",
            "justifyContent": "flex-start",
            "marginTop": 10,
          }
        }
        testID="Button__View"
      >
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 3.142857142857143,
              },
              {
                "width": "auto",
              },
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginLeft": 5,
                "opacity": 1,
              }
            }
            testID="Dialog__Button"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#007aff",
                  "borderRadius": 3.142857142857143,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 12.571428571428571,
                  "paddingHorizontal": 14.571428571428571,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#007aff",
                    "fontSize": 15,
                    "fontWeight": "500",
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Close
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`<InfoDialog /> snapshots renders correctly with title and children 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  replaceTheme={[Function]}
  testID="Internal__Overlay"
  transparent={true}
  updateTheme={[Function]}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, .4)",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    testID="RNE__Overlay__backdrop"
  />
  <View
    pointerEvents="box-none"
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="RNE__Overlay__Container"
  >
    <View
      style={
        {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderRadius": 3,
          "padding": 20,
          "shadowColor": "rgba(0, 0, 0, .3)",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowRadius": 4,
          "width": "75%",
        }
      }
      testID="RNE__Overlay"
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "600",
            "marginBottom": 10,
          }
        }
        testID="Dialog__Title"
      >
        Test Title
      </Text>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#242424",
            "fontFamily": undefined,
            "fontSize": 23.57142857142857,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "marginBottom": 0,
            "textAlign": "left",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
          }
        }
      >
        Test Content
      </Text>
      <View
        style={
          {
            "flexDirection": "row-reverse",
            "flexWrap": "wrap",
            "justifyContent": "flex-start",
            "marginTop": 10,
          }
        }
        testID="Button__View"
      >
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 3.142857142857143,
              },
              {
                "width": "auto",
              },
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginLeft": 5,
                "opacity": 1,
              }
            }
            testID="Dialog__Button"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#007aff",
                  "borderRadius": 3.142857142857143,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 12.571428571428571,
                  "paddingHorizontal": 14.571428571428571,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#007aff",
                    "fontSize": 15,
                    "fontWeight": "500",
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Close
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
